EpikCord/client/http.py

```diff
@@ -166,28 +211,57 @@ async def set_bucket(
         response: aiohttp.ClientResponse,
         major_parameters: Optional[MajorParameters] = None
     ) -> Union[Bucket, TopLevelBucket]:
+        """Sets the bucket for the request.
+
+        Parameters
+        ----------
+        response: aiohttp.ClientResponse
+            The response of the request.
+        major_parameters: Optional[MajorParameters]
+            The major parameters of the request.
+        
+        Returns
+        -------
+        Union[Bucket, TopLevelBucket]
+            The bucket for the request.
+        """
         url = response.url
         method = response.method
         bucket_key: str = f"{method}:{url}"
 
+        bucket: Optional[Union[Bucket, TopLevelBucket]] = None
+
         if major_parameters:
             bucket = TopLevelBucket(
-                **major_parameters.major_parameters
+                channel_id=major_parameters.channel_id,
+                guild_id=major_parameters.guild_id,
+                webhook_id=major_parameters.webhook_id,
+                webhook_token=major_parameters.webhook_token
```

Comment:
Just Make `TopLevelBucket` Inherit from `MajorParameters`

Reply:
That makes no sense though; MajorParameters is a class that groups all of the major parameters into a single class, a TopLevelBucket is a bucket used for handling rate limits.

Reply:
And the `TopLevelBucket` uses all the attributes of the `MajorParameters` in order to work. 

Inheritance *a mechanism where you can to derive a class from another class for a hierarchy of classes that **share a set of attributes and methods***

Reply:
I don't like the hierarchy tree for that though.

```
MajorParameters
            |
            |
            |
  TopLevelBucket

Reply:
Then use composition